includes/classes/class-woo-civi-admin-migrate.php

```diff
@@ -735,12 +734,33 @@ public function product_process( $product_id, $member_active ) {
 			delete_post_meta( $product_id, '_civicrm_contribution_type' );
 		}
 
+		// get all variation products 
+		$variation_product_id = [];
+		$args = array(
+			'post_type' => 'product_variation',
+			'post_parent' => $product_id,
+			'suppress_filters' => false
+		);
+		$all_variations = get_posts($args);
+		foreach ($all_variations as $result) {
+			array_push($variation_product_id, $result->ID);
+		}
+
 		// Does the Product have meta without the leading underscore?
 		$financial_type_id = get_post_meta( $product_id, 'woocommerce_civicrm_financial_type_id', true );
 
 		// When it does, update the proper meta and remove old.
-		if ( ! empty( $financial_type_id ) || $financial_type_id === 0 ) {
+		if ( ! empty( $financial_type_id ) || $financial_type_id === 0 ) {	
 			update_post_meta( $product_id, '_woocommerce_civicrm_financial_type_id', $financial_type_id );
+			update_post_meta( $product_id, '_woocommerce_civicrm_entity_type', 'civicrm_contribution' );
+			if ( $variation_product_id != NULL ) {
+				update_post_meta( $product_id, '_wpcv_wci_variable_civicrm_entity_type', 'civicrm_contribution' );
+				foreach ($variation_product_id as $variation_product) {
+					update_post_meta($variation_product, '_wpcv_wci_variable_membership_financial_type_id', $financial_type_id);
+					update_post_meta($variation_product, '_wpcv_wci_variable_contribution_financial_type_id', $financial_type_id);
+					update_post_meta($variation_product, '_wpcv_wci_variable_participant_financial_type_id', $financial_type_id);
```

Comment:
@lukeyangjma do we need to be updating each option not just `variable_contribution`?

Reply:
@seamuslee001 The entity types for all three (membership, contribution, and participants) all have the option to set the financial type for a products variations.

Membership:
<img width="800" alt="Screen Shot 2022-07-18 at 7 37 42 AM" src="https://user-images.githubusercontent.com/105511138/179503403-4e2c452b-9181-45f8-909a-c3a7f5cbcb7b.png">

Contribution:
<img width="813" alt="Screen Shot 2022-07-18 at 7 38 08 AM" src="https://user-images.githubusercontent.com/105511138/179503495-21566b42-cc41-46e0-b1ce-47c9f70ae338.png">

Partici
<img width="807" alt="Screen Shot 2022-07-18 at 7 37 23 AM" src="https://user-images.githubusercontent.com/105511138/179503513-8c6c98f0-c3a5-4b90-93f9-74e653a3b11d.png">
pant: